### ISSUE TITLE
clean_message() can strip comments from code blocks 

### ISSUE THREAD

**Issue by jwilk** [sociomantic-tsunami/git-hub#187]
Sometimes I want to paste Python code snippets into fenced code blocks.
But `clean_message()` removes all lines starting with <tt># </tt>, and that includes Python comments.
Now, I realize that this documented behavior (modulo bug #186), but it's rather surprising and there's no obvious work-around.
Please make `clean_message()` less zealous somehow.

**Comment by llucax** [comment]
I wonder how would this work in a regular git message... (although I acknowledge that is a different case, as it might be more common to write code in PR description that in commit messages).

I'm open to suggestion on how to solve this. An option would be to escape `#` with `\#` or something. Another option is not to include help on formatting and just use the message written by the user verbatim, but it will be quite a (breaking) change. Another option might be to have some sort of begin of verbatim message marker.

All those solutions seem quite ugly to me.

**Comment by jwilk** [comment]
>I wonder how would this work in a regular git message... (although I acknowledge that is a different case, as it might be more common to write code in PR description that in commit messages).

I wouldn't use fenced code block (or any other Markdown inventions...) in a commit message. Instead, I would indent the code with spaces.
(git doesn't strip indented `#`; git-hub does at the moment and it probably shouldn't.)

But if you really need to put leading `#` in a commit message, you can use `--cleanup=scissors` or `--cleanup=verbatim`.

>I'm open to suggestion on how to solve this. An option would be to escape `#` with `\#` or something.

Eww, ugly. Also, probably wouldn't help me much, because I would forget to escape it every time...

>Another option is not to include help on formatting and just use the message written by the user verbatim, but it will be quite a (breaking) change.

It wouldn't go this way. (I hope to get #166 fixed!)

>Another option might be to have some sort of begin of verbatim message marker.

:+1: We could use the marker from git's `--cleanup=scissors`, i.e.:
```
# ------------------------ >8 ------------------------
```

Yet another option is to parse Markdown, and strip only comments outside code blocks, but that would be probably very brittle...

**Comment by llucax** [comment]
OK, accepting a `--cleanup` option and behaving exactly like git (even using git's `commit.cleanup` and `code.commentChar`) seem to be the most sensible option for me.

**Comment by llucax** [comment]
Just for reference:
```
       --cleanup=<mode>
           This option determines how the supplied commit message should be cleaned up before committing. The <mode> can be strip, whitespace, verbatim, scissors or
           default.

           strip
               Strip leading and trailing empty lines, trailing whitespace, commentary and collapse consecutive empty lines.

           whitespace
               Same as strip except #commentary is not removed.

           verbatim
               Do not change the message at all.

           scissors
               Same as whitespace, except that everything from (and including) the line "# ------------------------ >8 ------------------------" is truncated if the
               message is to be edited. "#" can be customized with core.commentChar.

           default
               Same as strip if the message is to be edited. Otherwise whitespace.

           The default can be changed by the commit.cleanup configuration variable (see git-config(1)).
```

**Comment by daniel-zullo** [comment]
Addressed in #262. Please feel free to re-open it if that is not the case.
